src/mono/wasm/runtime/marshal-to-js.ts

```diff
@@ -364,6 +407,7 @@ function _marshal_js_object_to_js(arg: JSMarshalerArgument): any {
     }
     const js_handle = get_arg_js_handle(arg);
     const js_obj = mono_wasm_get_jsobj_from_js_handle(js_handle);
+    mono_assert(js_obj, "Expected JS object instance");
```

Comment:
A better message for this might be good if we expect it to ever be hit in production